src/scripts/reducers/ModalReducer.js

```diff
@@ -4,19 +4,21 @@ import ModalRecord from 'records/ModalRecord'
 import { OPEN_MODAL, CLOSE_MODAL } from 'constants/ActionConstants'
 
 import type { Action } from 'types/ApplicationTypes'
+import type { ModalPayload } from 'types/ModalTypes'
 
 const reducer = {
-  [OPEN_MODAL]: (state: ModalRecord, action: Action<string>) => {
-    return state.merge({
-      modalContent: action.payload,
-      previousModalContent: state.modalContent,
-      showModal: true
+  [OPEN_MODAL]: (state: ModalRecord, action: Action<ModalPayload>) => {
+    return state.withMutations((mutableState: ModalRecord) => {
+      mutableState.merge({
+        name: action.payload.modal,
+        previousModalName: state.get('name'),
+        showModal: true
+      })
+      mutableState.set('modalProps', action.payload.modalProps)
```

Comment:
If we include `modalProps` in `merge` then they will automatically be converted into an immutable record. This is a case where we don't need/really want to convert something to an immutable in our store.